Sync: Support sync connections to multiple sites

epeicher opened the pull request:
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "Fixes" keyword and use "Related to" instead.
-->

## Related issues

- Resolves STU-402

## Proposed Changes

- Allow connecting multiple sites behind the feature flag by enabling the `Connect site` button .
- Use the connected sites from the section to avoid duplicates.


||Before | After|
|-------|-------|------|
|Single site|![CleanShot 2025-04-30 at 13 55 40@2x](https://github.com/user-attachments/assets/b2cb1f71-dd7c-481b-892b-ea9650c11c99)| ![CleanShot 2025-04-30 at 13 55 09@2x](https://github.com/user-attachments/assets/3b3189ef-b95b-43fc-8a1c-b6a1f72fe0fe)| 
|Multiple sites|      N/A |   ![CleanShot 2025-04-30 at 13 55 14@2x](https://github.com/user-attachments/assets/77e20d28-2784-489f-ac55-9b7578321fb0)|
|Many sites|N/A|![CleanShot 2025-04-30 at 14 16 15@2x](https://github.com/user-attachments/assets/77db2934-a133-4148-98d6-c365d7d6605e)|
|No sites (should be similar)|![CleanShot 2025-04-30 at 14 10 58@2x](https://github.com/user-attachments/assets/c9d06619-b37b-4fe2-ac15-8f0909da947b)|    ![CleanShot 2025-04-30 at 14 16 31@2x](https://github.com/user-attachments/assets/75321deb-8199-44cc-903d-f441747bf4ff)|



## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

- Apply this patch
- Run `STUDIO_PRESSABLE_SYNC=true npm start` to run Studio with the feature flag enabled
- Go to the Sync tab and Connect one site
- Connect another site
- Ensure there are no unexpected errors
- Test Pulling and Pushing the sites
- Test the same using multiple Pressable (Jetpack connected) sites
* Stop Studio and start it again without the feature flag by running `npm start`
* Select a site without Connected sites, or disconnect the sites in a existing site
* Connect that site to any site
* Ensure you cannot see the `Connect site` button anymore


## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [x] Have you checked for TypeScript, React or other console errors?

epeicher committed:
Sync: Allow to sync connect with multiple sites

epeicher committed:
Use a different container for the Connect button

This keeps the Description component focused to its original behavior

epeicher committed:
Use connectedSites from section to avoid duplicates

katinthehatsite reviewed (approved):
I did not spot any specific regressions on my end, everything worked as expected and I was able to connect multiple sites. One thing that I was not sure about is if the button to connect should be secondary or primary but it seems like a design decision.

epeicher commented:
Thanks @katinthehatsite for the review! 
> One thing that I was not sure about is if the button to connect should be secondary or primary but it seems like a design decision.

That's an important point, I added `secondary` for the `Connect site` button for the subsequent connected sites because I saw it in the [linked task](https://linear.app/a8c/issue/STU-402/studio-sync-lets-support-connections-to-multiple-sites-ui). @sejas, could you confirm if that is expected?

sejas commented:
> That's an important point, I added secondary for the Connect site button for the subsequent connected sites because I saw it in the [linked task](https://linear.app/a8c/issue/STU-402/studio-sync-lets-support-connections-to-multiple-sites-ui). @sejas, could you confirm if that is expected?

Great question. I asked Marina on STU-402. I think it should be primary.

sejas reviewed (commented):
@epeicher , the code looks great! thank you!

Would be possible to reduce the spacing of the first element? See RToz6tIuQ7nlZrikBte4GU-fi-7953_459642
Let's also add/update some tests. At least to confirm the button appears even if there are connected sites.

As a stretch goal. Would be possible to add a blur effect in the bottom bar and reduce its padding top? If it's not an easy task, we can create a separate issue to handle it.

**FIGMA vs CODE**
![Screenshot 2025-04-29 at 19 23 42](https://github.com/user-attachments/assets/a14166a1-5d34-41ce-8e16-ae56d15647a6)

epeicher commented:
Thanks @sejas for the review and the suggestions! 

> Would be possible to reduce the spacing of the first element? See RToz6tIuQ7nlZrikBte4GU-fi-7953_459642

Sure thing, thanks for the link to the design doc.

> Let's also add/update some tests. At least to confirm the button appears even if there are connected sites.

👍 

> As a stretch goal. Would be possible to add a blur effect in the bottom bar and reduce its padding top? If it's not an easy task, we can create a separate issue to handle it.

Great suggestion! I will have a look at it, if I'm not able to complete it today I will create a task for it

ivan-ottinger commented:
> Great suggestion! I will have a look at it, if I'm not able to complete it today I will create a task for it

I haven't looked much further, but maybe we could reuse the styling which we have been using for the Previews tab:

![Markup on 2025-04-30 at 10:34:25](https://github.com/user-attachments/assets/a238b67a-ba22-4dd0-83f6-1cfeabb47603)

epeicher committed:
Remove extra space in Sync view list

epeicher committed:
Add blur effect to Connect site button area

Change the variant to primary

epeicher committed:
Add some tests to check the existance of Connect site button in Sync tab

epeicher commented:
> maybe we could reuse the styling which we have been using for the Previews tab:

Thanks @ivan-ottinger! I have reused some styles from that component, which has saved me considerable time.

@sejas, I have added your suggestions to the commits [44af572](https://github.com/Automattic/studio/pull/1291/commits/44af5721712d30a42786fa0c84ff34130b2b3d82), [69039d2](https://github.com/Automattic/studio/pull/1291/commits/69039d2ee4c311f7c8af8644b94c18fa534007cb) and [d704ae9](https://github.com/Automattic/studio/pull/1291/commits/d704ae9a60ff4940a5b8699d342a263152574374), they can be reviewed independently. I have changed the variant of the button to `primary` because it looked better with the blur effect. I have updated the description with screenshots. This is ready for another review.

@Marinatsu , could you please review the designs? I have added screenshots to the description for convenience.

sejas reviewed (approved):
Works great! I like how it's consistent with Preview tabs and the button is at the bottom until it "overflows" and get's sticky. Thanks for adding the blur effect.

https://github.com/user-attachments/assets/961c0397-64e0-4f61-a4b4-6bf8c030f206

ivan-ottinger reviewed (approved):
Nice! The proposed changes look good and the list is rendering correctly even when RTL locale is set. I did not observe any issues.

![Markup on 2025-05-02 at 16:24:56](https://github.com/user-attachments/assets/518585ba-3c91-47d2-8847-ce9c8d785a46)

Blur is working great and the `Connect site` button gets disabled when there's no internet.

sejas committed:
Merge branch 'trunk' of github.com:Automattic/studio into stu-402-studio-sync-lets-support-connections-to-multiple-sites-ui

sejas committed:
Restore the space between the logo title and the list of sites

sejas reviewed (approved):
I've noticed a the logo had less bottom space. I've fixed it on effc1256cde01b69e9443096120a20df959b2ffc

**BEFORE**

![before-logo-space](https://github.com/user-attachments/assets/7fed28c0-1b40-41f3-9690-ae64d496335a)

**AFTER**

![after-logo-space](https://github.com/user-attachments/assets/c6057c5b-f5d0-4c7d-bfa6-80c69859c321)

ivan-ottinger commented:
> I've noticed a the logo had less bottom space. I've fixed it on https://github.com/Automattic/studio/commit/effc1256cde01b69e9443096120a20df959b2ffc

Good catch!
